Oop№5   polymorphism

LarionovKirill opened the pull request.

LarionovKirill committed:
Feature(DelivaryTime.cs): создал перечисление в интервалами времени доставки.
Feature(PriorityOrder.cs): создал класс приоритетных заказов.

LarionovKirill committed:
Feature(model): везде добавил [Serialazible]
Feature(EnumGetter.cs): добавил сервисный класс для возвращения пересисления.

LarionovKirill committed:
Feature(CustomerTab.cs): изменил создание заказа по приоритетности.
Feature(Idgeneraor.cs): исправил XML.
Feature(OrderGenerator.cs): добавил генератор для тестирования.

LarionovKirill committed:
Fix(Model.): убрал из классов пустые поляи и заменил их на авто свойства.
Feature(OrderTab.cs): добавил возможность менять время доставки и видимость поля для приоритетных заказов.

LarionovKirill committed:
Fix(): заменил double на decimal

LarionovKirill committed:
Добавил диаграмму

LarionovKirill committed:
Merge branch 'OOP№3---Aggregation' into OOP№4---Inheritance

LarionovKirill committed:
Fix(project): Сделал перенос с 3 лабы.

LarionovKirill committed:
Маленькие изменения в верстке.

LarionovKirill committed:
Feature(PersentDiscount.cs): реализована логика класса.
Feature(PointsDiscount.cs): реализована логика класса.

LarionovKirill committed:
Feature(): добавлена промежуточная логика и изммененны некоторые подключения пространства имен

LarionovKirill committed:
Feature(AddDiscountForm.cs): Добавил форму для добавления скидки.
feature(OrdersTab.cs): Добавил форму интерфеус для взаимодействия со скидками.
Fix(Customer.cs): Испраивл порядок методов в классе.
Fix(PersentDiscount.cs): добавил конструктор с параметрами.

LarionovKirill committed:
Fix(Discounts): исправил методы подсчета скидки.
Fix(CartsTab.cs): Исправил добавление заказа.
Fix(CustomerTab.cs): Исправил обновление информации.
Refactoring(OrdersTab.cs): Мелкие изменения оформления.

LarionovKirill committed:
Fix(CustomerGenerator.cs): Переделал структуру генератора.
Fix(OrderTab.cs): Избавился от дублирования кода.
Refactoring(OrderTab.cs): добавил разделение между строками.

LarionovKirill committed:
Fix(ItemGenerator.cs): исправил генератор товаров.

LarionovKirill committed:
Fix(OrdersTab.cs): изменил размер таблицы.

LarionovKirill committed:
Refactoring(OrdersTab.cs): изменил пользовательский интерфейс с splitter

LarionovKirill committed:
Fix(EnumGetter.cs): исправил грамматическую ошибку.
Fix(OrderTab.cs): немного уменьшил вложенность кода.

LarionovKirill committed:
Refactoring(OrdersTab.cs): убрал обращение к пространству имен.

LarionovKirill committed:
Refactoring(ItemTab.cs): Добавил отступы.
Refactoring(CartsTab.cs): Добавил отступы.
Recatoring(MainForm.cs): Добавил отступы.
Fix(MainForm.cs): Убрал лишнее поле.

LarionovKirill committed:
Merge branch 'OOP№4---Inheritance' into OOP№5---polymorphism

LarionovKirill committed:
Update .gitignore

LarionovKirill committed:
Исправил некоторые недочеты после merge

LarionovKirill committed:
Промежуточный коммит

LarionovKirill committed:
feature(Diagrams):Добавил Uml диаграмму.
Fix(orderTab.cs): исправил баг с дублированием строк.

LarionovKirill committed:
Delete ObjectOrientedPractics/.vs/OOP/v16 directory

LarionovKirill committed:
Delete ObjectOrientedPractics/OOP/bin/Debug directory

LarionovKirill committed:
Delete ObjectOrientedPractics/OOP/obj directory

LarionovKirill committed:
Delete .gitignore

LarionovKirill committed:
Delete .gitignore

LarionovKirill committed:
Delete .gitignore

LarionovKirill committed:
Fix(OrdersTab.cs): уменьшил вложенность с прошлой лабы.

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Discounts/PercentDiscount.cs:
```diff
@@ -0,0 +1,142 @@
+    /// <summary>
+    /// Класс категорной скидки.
+    /// </summary>
+    public class PercentDiscount: IDiscount
```
До и после двоеточия должен стоять пробел.

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Discounts/PercentDiscount.cs:
```diff
@@ -0,0 +1,142 @@
+                _amountSpent = value;
+            }
+        }
+        /// <summary>
```
Разделитель

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Discounts/PercentDiscount.cs:
```diff
@@ -0,0 +1,142 @@
+        /// списка продуктов по определенной категории.
+        /// </summary>
+        /// <param name="items">Список продуктов.</param>
+        /// <returns></returns>
```
XML комментарий к возвращаемому значению (не хватает в нескольких местах).

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Discounts/PercentDiscount.cs:
```diff
@@ -0,0 +1,142 @@
+    /// <summary>
+    /// Класс категорной скидки.
+    /// </summary>
+    public class PercentDiscount: IDiscount
```
В компании есть договорённость по очередность расположения полей в классах:

**Все содержимое классов должно группироваться следующим образом:**
1. Константы
2. Перечисления
3. Внутренние классы
4. Делегаты
5. Поля
6. Свойства
7. События
8. Конструкторы
9. Публичные методы
10. Защищенные методы
11. Приватные методы

**Свойства необходимо сортировать по модификатору доступа:**
1. Приватные
2. Приватные на set, публичные на get
3. Обычные
4. Только на get

Видео на тему почему стоит соблюдать договорённости по оформлению кода и прочего: https://youtu.be/KpohpURAKUQ

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Discounts/PointsDiscount.cs:
```diff
@@ -0,0 +1,107 @@
+
+namespace OOP.Model.Discounts
+{
+    public class PointsDiscount : IDiscount
```
XML комментарий к классу.

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CustomerTab.cs:
```diff
@@ -247,13 +268,105 @@ private void CancelButton_Click(object sender, EventArgs e)
+        /// <summary>
+        /// Метод переводит на форму с добавлением формы.
+        /// </summary>
+        private void AddDiscountButton_Click(object sender, EventArgs e)
```
https://github.com/LarionovKirill/Programming/assets/50287099/efae65dc-f757-4ddc-b66c-adfbe58e94d5

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/Model/DeliveryTime.cs:
```diff
@@ -0,0 +1,23 @@
+    /// <summary>
+    /// Время доставки.
+    /// </summary>
+    [Serializable]
```
Вроде посмотрел везде и сериализации данных у тебя нигде нету. Если это так, то зачем тебе этот атрибут?

LarionovKirill replied:
Вернусь к той проблеме. В ней описано, что класс не помечен как сериализуемый. Я на всякий случай пометил все и добавил атрибуты про которые ты писал в прошлом пулле. Сейчас я убрал [Serializable]  и вроде он работает, но посмотрим, если вернется ошибка, то верну на нужные классы.
![image](https://github.com/LarionovKirill/Programming/assets/122218338/553ff273-b194-4d0a-983d-cc99a9018514)

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CartsTab.cs:
```diff
@@ -1,12 +1,10 @@
+using OOP.Model.Enums;
+using OOP.Model.Discounts;
 
 namespace OOP.View.Tabs
```
Чекбоксы на скидку не отрабатывают.
https://github.com/LarionovKirill/Programming/assets/50287099/109f4893-36a0-42d0-a30f-089110ce95d6

LarionovKirill replied:
Исправил

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CartsTab.cs:
```diff
@@ -147,7 +161,75 @@ private void ClearCartButton_Click(object sender, EventArgs e)
+        /// <param name="currentCustomer">Переданный пользователь.</param>
+        private void FillDiscounts(Model.Customer currentCustomer)
+        {
+            var a = 0;
```
Название переменной не отражает её сути. Это не математическая формула где такие обозначения уместны.

LarionovKirill replied:
Исправил

LarionovKirill committed:
fix(PercentDiscount.cs): добавил пробел при наследовании.

LarionovKirill committed:
Fix(PercentDiscount.cs): добавил разделение.

LarionovKirill committed:
Fix(PercentDiscount.cs): Добавил описание возврата в XML - комментарии.
Fix(PointsDiscount.cs): Добавил описание возврата в XML - комментарии.

LarionovKirill committed:
Fix(PointsDiscount.cs): добавил XML - комментарий к классу.

LarionovKirill committed:
Fix(Model): Убрал атрибут [Serializable] со все классов.
Fix(Item.cs):Изменил структуру класса по правилам, которые описаны.
Fix(Customer.cs):Изменил структуру класса по правилам, которые описаны.
Fix(Discounts):Изменил структуру класса по правилам, которые описаны.

LarionovKirill committed:
Fix(AddDiscountForm.cs): исправил баг в подтверждением пустой скидки.

LarionovKirill committed:
Feature(AddDiscountForm.cs): добавил стиль у ComboBox как DropDownList

LarionovKirill committed:
Fix(CartsTab.cs): исправил имя локальной переменной на осмысленное.

LarionovKirill committed:
Fix(CartsTab.cs): исправил баг с зачислением скидки и реагированием checkbox.
fix(OrdersTab.cs): исправил баг с итоговой стоимостью.

LarionovKirill committed:
Feature(Diagrams): добавил второй вариант диаграммы.
